File: game/src/main/java/me/firecloud/game/rule/OrNode.java
/**
 * 
 */
package me.firecloud.game.rule;

import java.util.List;

/**
 * @date 2012-10-12
 * 
 */
public class OrNode
    implements Node {

  protected List<Node> nextNodes;

  /*
   * (non-Javadoc)
   * 
   * @see me.firecloud.game.rule.Node#handle(me.firecloud.game.rule.Event)
   */
  public
      Node handle(Event event,GameContext gameContext) throws RuleException {
    for (Node node : nextNodes) {
      try {
        Node nextNode = node.handle(event,gameContext);
        if (node != null) {
          return nextNode;
        }
      } catch (RuleException ex) {

      }
    }

    return null;
  }

}

File: card/src/main/scala/me/firecloud/card/rule/OrNode.scala
/**
 *
 */
package me.firecloud.card.rule

/**
 * @author kkppccdd
 * @email kkppccdd@gmail.com
 * @date Oct 19, 2012
 *
 */
class OrNode(val nodes: List[Node]) extends Node {

  override def handle(gameContext: GameContext, event: Event): Node = {
    for (node <- nodes) yield {
      var result = node.handle(gameContext, event)
      if (result != null) {
        return result
      }
    }

    return null

  }

}